ts/Extensions/Annotations/NavigationBindings.ts

```diff
@@ -1486,6 +1487,52 @@ setOptions({
         }
     }
 });
+addEvent(H.Chart, 'render', function (): void {
+    const chart = this,
+        navigationBindings = chart.navigationBindings,
+        disabledClassName = 'highcharts-disabled-btn';
+
+    if (chart && navigationBindings) {
+        // Check if the buttons should be enabled/disabled based on
+        // visible series.
+
+        let buttonsEnabled = false;
+        chart.series.forEach(function (series): void {
+            if (!series.navigatorSeries && series.visible) {
+                buttonsEnabled = true;
+            }
+        });
+
+        objectEach(
+            navigationBindings.boundClassNames,
+            function (value: Highcharts.NavigationBindingsOptionsObject, key: string): void {
+                // Get the HTML element coresponding to the
+                // className taken from StockToolsBindings.
+
+                if (chart.stockTools && chart.stockTools.toolbar) {
+                    const buttonNode = chart.stockTools.toolbar.querySelectorAll('.' + key);
+
+                    if (buttonNode) {
+                        if (!buttonsEnabled && !value.alwaysEnabled) {
```

Comment:
Can we move `value.alwaysEnabled` to the line 1512? Whole checking DOM, and classnames operations are unnecessary when buttons should be always enabled.